x-pack/plugins/triggers_actions_ui/public/application/components/json_editor_with_message_variables.tsx

```diff
@@ -63,10 +75,35 @@ export const JsonEditorWithMessageVariables: React.FunctionComponent<Props> = ({
     onDocumentsChange(convertToJson(newValue));
   };
 
-  const onClickWithMessageVariable = (_value: any) => {
-    setCursorPosition(_value);
+  const registerEditorListeners = useCallback(() => {
+    const editor = editorRef.current;
+    if (!editor) {
+      return;
+    }
```

Comment:
Lets think about what happens if the editor is null here...
We don't want _nothing_ to happen, as that leaves users confused and unable to progress.
An explicit error is preferred as they can report it.

Reply:
Yea sure I'm open to suggestions, although this is a guard for the initial register call since we're reusing this method in the `useEffect`, which will call this method once before we get a reference to the editor.

We could add a console.error/warn, or maybe split it into 2 cases, 1 where the editor exists, another when it doesn't exist, just to be more clear

Reply:
Yea I wonder if we can show something in the UI in this scenario - some kind of `<EuiCallout>` to indicate what the problem and what the user can do about it (I'm not sure if we even know this TBH)

Reply:
I'd opt for a UX that the user sees so that they know something went wrong and they can report it.